### PR TITLE
Refactor activity namespace and improve views

### LINKED ISSUE #7: Refactor activity namespace
Arne explained to me that transformations of data are better implemented at the edges (in this case, the `activities.handlers` namespace) to avoid improper data from rolling deep into the application and to make the consequences of calling a specific handler more visible. Also, the current namespace abstracts operations over activities from HTTP requests and databases—which are two different domains. This is a warning sign that the code probably needs refactoring.

### PR DESCRIPTION
**Pull request by loopified**
Closes #7

### CODE REVIEW

File: src/activities/activity.clj [hunk 1]
```diff
@@ -17,39 +17,51 @@
...
-  namespaced activity map."
-  [{{:keys [title description datetime duration capacity]} :params
-    {:keys [identity]} :session}]
-  (let [new-uuid (UUID/randomUUID)
-        inst     (utils/datetime->inst datetime)
+(defn new-activity
+  "Takes a map of activity attributes and returns a properly namespaced and
+  conformed activity map with some defaults filled in."
+  [{:keys [uuid creator title description datetime duration capacity]}]
+  (let [inst     (utils/datetime->inst datetime) ;There is a NullPointerException coming from this line
```
Comment (Heliosmaster) [inline src/activities/activity.clj]: remove this rogue comment

File: src/activities/activity.clj [hunk 2]
```diff
@@ -17,39 +17,51 @@
...
-(defn req->activity
-  "Takes a request containing an activity id in the path and returns the
-  associated entity in the database."
-  [req]
-  (let [db       (crux/db (:crux req))
-        id       (get-in req [:path-params :id])
-        uuid     (UUID/fromString id)
-        activity (crux/entity db uuid)]
-    activity))                          ;TODO validate with spec
+(defn find-activities
```
Comment (Heliosmaster) [inline src/activities/activity.clj]: i would name it `list-issues`

File: src/activities/handlers.clj [hunk 3]
```diff
@@ -49,71 +49,82 @@
...
+  (let [new-uuid   (UUID/randomUUID)
+        creator    (:identity session)
+        attributes (assoc params :uuid new-uuid :creator creator)
+        activity   (activity/new-activity attributes)
+        path       (path router :activities.system/activity {:id new-uuid})]
+    (if (s/valid? :activities/activity activity)
+      (do (submit-tx crux [[:crux.tx/put activity]])
+          {:status  303
+           :headers {"Location" path}})
+      (s/explain :activities/activity activity)))) ;should it throw?
```
Comment (Heliosmaster) [inline src/activities/handlers.clj]: you probably should return a normal response. Something like `{:status 400 :body (s/explain ... )}` (look up the most appropriate http code)

File: src/activities/handlers.clj [hunk 4]
```diff
@@ -49,71 +49,82 @@
...
+        {:keys [:crux.db/id
+                ::activity/creator]} current-activity
+        new-activity          #p (merge params {:uuid id :creator creator})
+        updated-activity      (activity/new-activity new-activity)]
+    (submit-tx crux [[:crux.tx/put updated-activity]])
     {:status  301
-     :headers {"Location" (path req :activities.system/activity {:id activity-id})}}))
+     :headers {"Location" (path router :activities.system/activity {:id activity-id})}}))
+
+;;         _ (prn [:new-activity new-activity])
```
Comment (Heliosmaster) [inline src/activities/handlers.clj]: remove

File: src/activities/views.clj [hunk 5]
```diff
@@ -215,23 +203,19 @@
...
-        description (:activity/description activity)
-        inst        (:activity/date-time activity)
-        datetime    (-> inst
-                        (time/local-date-time (time/zone-id "UTC"))
-                        (time/truncate-to :minutes)
-                        (time/format))
-        duration    (:activity/duration activity)
-        capacity    (:activity/capacity activity)]
+  [{:keys [:crux.db/id ::a/title ::a/description ::a/date-time ::a/duration
+           ::a/capacity]} _ router]
```
Comment (Heliosmaster) [inline src/activities/views.clj]: don't use `_` unless it's an anonymous function. I would, in this case, use `_user-id` so you're still signaling what the field is and that is currently unused.

**Review by Heliosmaster — CHANGES_REQUESTED** [review]

File: src/activities/activity.clj (hunk 2 above)
Comment (loopified) [reply src/activities/activity.clj]: I believe you meant `list-activities`, right? If so, it's fixed.

File: src/activities/handlers.clj (hunk 3 above)
Comment (loopified) [reply src/activities/handlers.clj]: I have realized that the function had that error page but I removed it trying to get it into a working state a few commits ago and forgot to add it again. Thanks.

**Comment by loopified** [comment]
@Heliosmaster sorry for taking so long. I have fixed the issues but I'm not certain about the workflow on GitHub after I pushed the commits. If I understood this article (https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-request-reviews) correctly I should resolve the conversations since I have initiated the pull request, which I did.

Also, the code snippets didn't update automatically like we imagined and only displayed an "outdated" tag on top of them.

Please re-review :) Thank you again!

**Pull request merged**